src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2742,4 +2743,163 @@ namespace winrt::TerminalApp::implementation
+                {
+                    // Get the settings of the focused control and stash them
+                    auto controlSettings = activeControl.Settings().as<TerminalSettings>();
+                    // Make sure to recurse up to the root - if you're doing
+                    // this while you're currently previewing a SetColorScheme
+                    // action, then the parent of the control's settings is _the
+                    // last preview TerminalSettings we inserted! we don't want
+                    // to save that one!
+                    _originalSettings = controlSettings.GetParent();
+                    while (_originalSettings.GetParent() != nullptr)
+                    {
+                        _originalSettings = _originalSettings.GetParent();
+                    }
+                    // Create a new child for those settings
+                    TerminalSettingsCreateResult fake{ _originalSettings, nullptr };
+                    auto childStruct = TerminalSettings::CreateWithParent(fake);
+                    // Modify the child to have the applied color scheme
+                    childStruct.DefaultSettings().ApplyColorScheme(scheme);
+
+                    // Insert that new child as the parent of the control's settings
+                    controlSettings.SetParent(childStruct.DefaultSettings());
+                    activeControl.UpdateSettings();
+                }
+            }
+        }
+    }
+
+    // - Handler for the CommandPalette::PreviewAction event. The Command
+    //   Palette will raise this even when an action is selected, but _not_
+    //   committed. This gives the Terminal a chance to display a "preview" of
+    //   the action.
+    // - This will be called with a null args before an action is dispatched, or
+    //   when the palette is dismissed.
+    // - For any actions that are to be previewed here, MAKE SURE TO RESTORE THE
+    //   STATE IN `TerminalPage::_EndPreview`. That method is called to revert
+    //   the terminal to the state it was in at the start of the preview.
+    // - Currently, only SetColorScheme actions are preview-able.
+    // Arguments:
+    // - args: The Command that's trying to be previewed, or nullptr if we should stop the preview.
+    // Return Value:
+    // - <none>
+    void TerminalPage::_PreviewActionHandler(const IInspectable& /*sender*/,
+                                             const Microsoft::Terminal::Settings::Model::Command& args)
+    {
+        if (args == nullptr || args.Action() == nullptr)
+        {
+            _EndPreview();
+        }
+        else
+        {
+            switch (args.Action().Action())
+            {
+            case ShortcutAction::SetColorScheme:
+            {
+                _PreviewColorScheme(args.Action().Args().try_as<SetColorSchemeArgs>());
+                break;
+            }
+            }
+
+            // Other ideas for actions that could be preview-able:
```

Comment:
GH todo linked?

Reply:
Meh, I didn't file one but I'll do it

Reply:
https://github.com/microsoft/terminal/issues/9818